src/client/components/app/Transactions/RepayPositionTx.tsx

```diff
@@ -136,15 +147,36 @@ export const RepayPositionTx: FC<RepayPositionProps> = (props) => {
   const [haveFetched0x, setHaveFetched0x] = useState<boolean>(false);
   // const [zeroExWarning, setZeroExWarning] = useState<boolean>(false);
   const [isTrade, setIsTrade] = useState<boolean>(false);
+  const [costToClose, setCostToClose] = useState('0');
+  const [claimableTokenAddresses, setClaimableTokenAddresses] = useState(['']);
+  const [blockZeroExTrade, setBlockZeroExTrade] = useState(false);
   // TODO: replace usage in useFundsForClosing with appropriate value
 
+  useEffect(() => {
+    if (selectedPosition && selectedLine?.id) {
+      dispatch(LinesActions.getInterestAccrued({ contractAddress: selectedLine.id, id: selectedPosition.id }));
+    }
+  }, []);
+
   useEffect(() => {
     if (!selectedSellToken && !_.isEmpty(sourceAssetOptions)) {
-      dispatch(
-        TokensActions.setSelectedTokenAddress({
-          tokenAddress: sourceAssetOptions[0].address,
-        })
-      );
+      if (Object.keys(reservesMap).length !== 0) {
```

Comment:
can u refactor the ifs to be flatter/less confusing
```
if !selectedToken {
   if Object.keys(reservesMap).length !== 0
    ...blah
    return dispatch
  if !_.isEmpty(sourceAssetOptions) 
     ...blah
     return dispatch
```

should have the same result but less branching